acceptance/tests/resource/service/puppet_service_runs_puppet.rb

```diff
@@ -30,7 +30,7 @@
       assert_service_status_on_host(agent, 'puppet', {'ensure' => 'running'})
     end
 
-    retry_params = {:max_retries => 10,
+    retry_params = {:max_retries => 15,
```

Comment:
It's surprising to me that there is a noticeable slow down. I wouldn't expect calls to `LoadLibrary` to add several seconds, but perhaps ruby is scanning the `LOAD_PATH` instead of using `%WINDIR%`? Might be good to run `procmon.exe` and see what ruby is up to?

Reply:
I think this timer here was too short to begin with, for Windows at least.

On PR testing only windows10-64a failed, the other hosts passed, and when I ran the same thing locally with beaker the test passed as well.

I'll time this test before and after this PR to see how noticeable the change is.

Also, this will go away when we finish the work on PUP-10606 which extracts the shared FFI code into a separate module.

Reply:
Ok, weird stuff, I'm getting consistent times with and without this PR (the puppet service start time is somewhere around 9.5-10.5 seconds).

I wrote a small script to check out the load time of `puppet/util/windows` and you were right, it doesn't add much time:

```ruby
starting = Process.clock_gettime(Process::CLOCK_MONOTONIC)
require 'puppet/util/windows'
ending = Process.clock_gettime(Process::CLOCK_MONOTONIC)
puts ending - starting
```

```
C:\>ruby test.rb
0.8680698000002849
```

I just noticed that this timer is about waiting for the run to finish, so the service is started beforehand. Based on the code from this PR, the duration of `puppet agent -t` shouldn't change at all, so are these just infrastructure hiccups?